### PR TITLE
New version: Mega.MEGASync version 4.12.2.0

### PR DESCRIPTION
**Pull request by ypurpl**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/138222)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> 2024-02-07T04:28:01.3608103Z ##[error] Manifest Error: Duplicate installer entry found.

(Automated response - build 695.)

**Comment by stephengillie** [comment]
Automatic Validation ended with:
>  Installation failed with exit code -1978335226 2024-02-07 10:25:21.175 [CLI ] ShellExecute installer failed: 2

(Automated response - build 708.)

**Comment by stephengillie** [comment]
Hi @vvirtues,

> This PR's version number 4.12.2 has 3 parameters (sets of numbers between dots - major, minor, etc), which is less than the current manifest's version 4.11.0.0, which has 4 parameters.

Does this PR's version number **exactly** match the version reported in the Apps & features Settings page? (Feel free to attach a screenshot.)

(Automated response - build 720)

**Comment by vvirtues** [comment]
> Hi @vvirtues,
> 
> > This PR's version number 4.12.2 has 3 parameters (sets of numbers between dots - major, minor, etc), which is less than the current manifest's version 4.11.0.0, which has 4 parameters.
> 
> Does this PR's version number **exactly** match the version reported in the Apps & features Settings page? (Feel free to attach a screenshot.)
> 
> (Automated response - build 720)

uh
image

**Comment by stephengillie** [comment]
This application doesn't write version data to the expected locations:

image

The EXE properties show a matching version:

image

But this has an extra `.0`, which has as many parameters as the previous version.

File: manifests/m/Mega/MEGASync/4.12.2/Mega.MEGASync.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Created with Komac v1.11.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: Mega.MEGASync
+PackageVersion: 4.12.2
```
Comment (stephengillie) [inline manifests/m/Mega/MEGASync/4.12.2/Mega.MEGASync.installer.yaml:5]:
```suggestion
PackageVersion: 4.12.2.0
```
Please update file name and path to match this change.

File: manifests/m/Mega/MEGASync/4.12.2/Mega.MEGASync.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Created with Komac v1.11.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.5.0.schema.json
+
+PackageIdentifier: Mega.MEGASync
+PackageVersion: 4.12.2
```
Comment (stephengillie) [inline manifests/m/Mega/MEGASync/4.12.2/Mega.MEGASync.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 4.12.2.0
```
Please update file name and path to match this change.

File: manifests/m/Mega/MEGASync/4.12.2/Mega.MEGASync.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Created with Komac v1.11.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.5.0.schema.json
+
+PackageIdentifier: Mega.MEGASync
+PackageVersion: 4.12.2
```
Comment (stephengillie) [inline manifests/m/Mega/MEGASync/4.12.2/Mega.MEGASync.yaml:5]:
```suggestion
PackageVersion: 4.12.2.0
```
Please update file name and path to match this change.

**Comment by vvirtues** [comment]
Do I move the entire version to a new folder?

**Comment by vvirtues** [comment]
or do I remove the `.0` from certain manifest files?

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> 2024-02-09T21:48:25.9583906Z ##[error] One or more errors occurred. (Manifest: The manifest file path must match PackageIdentifier and PackageVersion properties from manifest. actual 'manifests\m\Mega\MEGASync\4.12.2', expected 'manifests\m\Mega\MEGASync\4.12.2.0') Manifest: The manifest file path must match PackageIdentifier and PackageVersion properties from manifest. actual 'manifests\m\Mega\MEGASync\4.12.2', expected 'manifests\m\Mega\MEGASync\4.12.2.0'

(Automated response - build 726.)

**Comment by stephengillie** [comment]
> Do I move the entire version to a new folder?

Or rename the folder.

**Comment by vvirtues** [comment]
haven't had time to look at it, will review over the weekend

**Comment by stephengillie** [comment]
Automatic Validation ended with:
>  2024-02-26 14:59:32.774 [CLI ] Installer [X64,nullsoft,User,en-US] not applicable: Machine is not compatible with X64

(Automated response - build 828.)

File: manifests/m/Mega/MEGASync/4.12.2.0/Mega.MEGASync.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+- silentWithProgress
+UpgradeBehavior: install
+AppsAndFeaturesEntries:
+- DisplayVersion: 4.12.2
+Installers:
+- Architecture: x64
+  InstallerUrl: https://mega.nz/MEGAsyncSetup32.exe
+  InstallerSha256: 73BCEE878DA54C3B37C2EE00F4B3E3070136EA1D2E3438A7299537FD5875AF61
+- Architecture: x86
+  InstallerUrl: https://mega.nz/MEGAsyncSetup64.exe
```
Comment (stephengillie) [inline manifests/m/Mega/MEGASync/4.12.2.0/Mega.MEGASync.installer.yaml:21]:
```suggestion
  InstallerUrl: https://mega.nz/MEGAsyncSetup64.exe
  InstallerSha256: 73BCEE878DA54C3B37C2EE00F4B3E3070136EA1D2E3438A7299537FD5875AF61
- Architecture: x86
  InstallerUrl: https://mega.nz/MEGAsyncSetup32.exe

```

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @vvirtues,

It looks like the architectures got swapped.

**Comment by vvirtues** [comment]
mightve updated

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> Installer Verification Analysis Context Information: 
 Specified hash doesn't match.
 Actual hash:   CCE0585C865A68C4C124FA450A82831F163C7FDC3569AFD966F1CB30F87A963D
 Expected hash: 73bcee878da54c3b37c2ee00f4b3e3070136ea1d2e3438a7299537fd5875af61
 Specified hash doesn't match.
 Actual hash:   73BCEE878DA54C3B37C2EE00F4B3E3070136EA1D2E3438A7299537FD5875AF61
 Expected hash: cce0585c865a68c4c124fa450a82831f163c7fdc3569afd966f1cb30f87a963d
 
 2/27/2024 5:06:52 PM
 
 ****************************************************************************************

(Automated response - build 835.)

File: manifests/m/Mega/MEGASync/4.12.2.0/Mega.MEGASync.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- interactive
+- silent
+- silentWithProgress
+UpgradeBehavior: install
+AppsAndFeaturesEntries:
+- DisplayVersion: 4.12.2
+Installers:
+- Architecture: x64
+  InstallerUrl: https://mega.nz/MEGAsyncSetup64.exe
+  InstallerSha256: 73BCEE878DA54C3B37C2EE00F4B3E3070136EA1D2E3438A7299537FD5875AF61
```
Comment (stephengillie) [inline manifests/m/Mega/MEGASync/4.12.2.0/Mega.MEGASync.installer.yaml:19]:
```suggestion

```

(Automated response - build 835.)

File: manifests/m/Mega/MEGASync/4.12.2.0/Mega.MEGASync.installer.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- Architecture: x86
+  InstallerUrl: https://mega.nz/MEGAsyncSetup32.exe
+
+  InstallerSha256: CCE0585C865A68C4C124FA450A82831F163C7FDC3569AFD966F1CB30F87A963D
```
Comment (stephengillie) [inline manifests/m/Mega/MEGASync/4.12.2.0/Mega.MEGASync.installer.yaml:23]:
```suggestion
  InstallerSha256: CCE0585C865A68C4C124FA450A82831F163C7FDC3569AFD966F1CB30F87A963D
- Architecture: x86
  InstallerUrl: https://mega.nz/MEGAsyncSetup32.exe
  InstallerSha256: 73BCEE878DA54C3B37C2EE00F4B3E3070136EA1D2E3438A7299537FD5875AF61
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
The installer hashes need to be swapped also. I forgot when I swapped the installers.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**